react-hot-button-component/src/hot-button.jsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+
+export default class HotButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+    this.state = { timesClicked: 0 };
+  }
+
+  handleClick() {
+    this.setState({ timesClicked: this.state.timesClicked + 1 });
+  }
+
+  render() {
+    if (this.state.timesClicked >= 18) {
+      return (
+        <button onClick={this.handleClick} className='white'>
+          Hot Button
+        </button>
+      );
+    } else if (this.state.timesClicked >= 15) {
+      return (
+        <button onClick={this.handleClick} className='yellow'>
+          Hot Button
+        </button>
+      );
+    } else if (this.state.timesClicked >= 12) {
+      return (
+        <button onClick={this.handleClick} className='orange'>
+          Hot Button
+        </button>
+      );
+    } else if (this.state.timesClicked >= 9) {
+      return (
+        <button onClick={this.handleClick} className='red'>
+          Hot Button
+        </button>
+      );
+    } else if (this.state.timesClicked >= 6) {
+      return (
+        <button onClick={this.handleClick} className='purple'>
+          Hot Button
+        </button>
+      );
+    } else if (this.state.timesClicked >= 3) {
+      return (
+        <button onClick={this.handleClick} className='blue'>
+          Hot Button
+        </button>
+      );
+    } else {
+      return (
+      <button onClick={this.handleClick}>
+        Hot Button
+      </button>
+      );
```

Comment:
Instead of creating several button elements it is better to use variables for the parts that change between each button. In this case only the buttons class changes. So a better approach would be:
```JSX
let btnClass = '';
if (this.state.timesClicked >= 18) {
  btnClass = 'white';
} else if (this.state.timesClicked >= 15) {
  btnClass = 'yellow';
} else if (this.state.timesClicked >= 12) {
  btnClass = 'orange';
} else if (this.state.timesClicked >= 9) {
  btnClass = 'red';
} else if (this.state.timesClicked >= 6) {
  btnClass = 'purple';
} else if (this.state.timesClicked >= 3) {
  btnClass = 'blue';
}

return (
  <button onClick={this.handleClick} className={btnClass}>
    Hot Button
  </button>
);
```